ambari-server/src/main/resources/common-services/HBASE/0.96.0.2.0/package/scripts/phoenix_queryserver.py

```diff
@@ -45,6 +47,9 @@ def start(self, env, upgrade_type=None):
     self.configure(env)
     phoenix_service('start')
 
+  @retry(times=3, sleep_time=5, err_class=Fail)
+  def post_start(self, env=None):
+    return super(PhoenixQueryServer, self).post_start(env)
```

Comment:
my worry is what happens if somebody adds some non-idempotent code into post_start() method? How about leaving some comment inside the method?

Reply:
Good point. However we're retying only if a specific exception was thrown. Hopefully it won't cause any problems, nevertheless I added a comment.